hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -171,14 +175,25 @@ def __init__(self, weights: Sequence[float]):
                 self.table.append((base, alternate, alternate_chance))
         self.table.sort()
 
-    def sample(self, data: "ConjectureData") -> int:
+    def sample(self, data: "ConjectureData", forced: Optional[int] = None) -> int:
         data.start_example(SAMPLE_IN_SAMPLER_LABEL)
-        base, alternate, alternate_chance = choice(data, self.table)
-        use_alternate = data.draw_boolean(alternate_chance)
+        forced_choice = (
+            None
+            if forced is None
+            else next((b, a, a_c) for (b, a, a_c) in self.table if forced in (b, a))
```

Comment:
no comment...ugly, but gets the job done. If I've missed an insight that renders this (and the `forced` in `choice`) unnecessary, let me know.